econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -955,58 +1031,107 @@ def _whatif_dict(self, X, Xnew, feature_index, y, alpha=0.1):
+        alpha : float in [0, 1], default 0.1
+            Confidence level of the confidence intervals displayed in the leaf nodes.
+            A (1-alpha)*100% confidence interval is displayed.
 
         Returns
         -------
+        y_new: DataFrame
+            The predicted outputs that would have been observed under the counterfactual features
+        """
+        return self._whatif_inference(X, Xnew, feature_index, y).summary_frame(alpha=alpha)
+
+    def _whatif_dict(self, X, Xnew, feature_index, y, alpha=0.1):
+        """
+        Get counterfactual predictions when feature_index is changed to Xnew from its observational counterpart.
+
+        Note that this only applies to regression use cases; for classification what-if analysis is not supported.
+
+        Parameters
+        ----------
+        X: array-like
+            Features
+        Xnew: array-like
+            New values of a single column of X
+        feature_index: int or string
+            The index of the feature being varied to Xnew, either as a numeric index or
+            the string name if the input is a dataframe
+        y: array-like
+            Observed labels or outcome of a predictive model for baseline y values
+        alpha : float in [0, 1], default 0.1
+            Confidence level of the confidence intervals displayed in the leaf nodes.
+            A (1-alpha)*100% confidence interval is displayed.
+        Returns
+        -------
         dict : dict
             The counterfactual predictions, as a dictionary
         """
 
-        inf = self.whatif(X, Xnew, feature_index, y)
+        inf = self._whatif_inference(X, Xnew, feature_index, y)
         props = self._point_props(alpha=alpha)
         res = _get_default_specific_insights('whatif')
         res.update([(key, self._make_accessor(attr)(inf).tolist()) for key, attr in props])
         return res
 
-    def _tree(self, is_policy, Xtest, feature_index, *, treatment_cost=0,
+    def _tree(self, is_policy, Xtest, feature_index, *, treatment_costs=0,
               max_depth=3, min_samples_leaf=2, min_impurity_decrease=1e-4, alpha=.1):
 
         result = self._safe_result_index(Xtest, feature_index)
         Xtest = result.X_transformer.transform(Xtest)
         if Xtest.shape[1] == 0:
             Xtest = None
         if result.feature_baseline is None:
-            treatment_names = ['low', 'high']
+            treatment_names = ['decrease', 'increase']
         else:
             treatment_names = [f"{result.feature_baseline}"] + \
                 [f"{lvl}" for lvl in result.feature_levels]
 
-        if len(treatment_names) > 2 and is_policy:
-            raise AssertionError("Can't create policy trees for multi-class features, "
-                                 f"but this feature has values {treatment_names}")
-
         TreeType = SingleTreePolicyInterpreter if is_policy else SingleTreeCateInterpreter
         intrp = TreeType(include_model_uncertainty=True,
```

Comment:
Since we are not surfacing the uncertainty I would either set this to False (as it severely impacts run time especially in forests) or more ideally, I would make `include_uncertainty` as an extra option, with default = False, in all the policy methods and set this equal to the passed `include_uncertainty` option

Reply:
Done